pamelalozano/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -43,23 +43,63 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-    ArrayList<String> comments = new ArrayList<>();  
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(3);
 
-    Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
-    PreparedQuery resultsQuery = datastore.prepare(query);
-    List<Entity> results = resultsQuery.asList(FetchOptions.Builder.withLimit(3));
+     String startCursor = request.getParameter("cursor");
+     String next = request.getParameter("next"); //if back this is null
```

Comment:
This could be a single parameter "paginationDirection" which could have a value of "up" or "down". The URL param would then be `?paginationDirection=up` for newer comments; `?paginationDirection=down` for older comments.